eclipse/che-theia
# Support more than one authentication session

Signed-off-by: Igor Vinokur <ivinokur@redhat.com>



### What does this PR do?
* Handle multiple GitHub authentication sessions.
* Add back the vscode built-in github plugin to be able to publish `https` repositories. 

### Screenshot/screencast of this PR



### What issues does this PR fix or reference?

fixes https://github.com/eclipse/che/issues/18227

### How to test this PR?

1. Setup [GitHub Identity provider](https://www.eclipse.org/che/docs/che-7/end-user-guide/configuring-github-oauth/)
2. Start a workspace with GitHub Pull Requests plugin and a project cloned from GitHub by `https` url.
You can use this devfile to start a workspace:
```
apiVersion: 1.0.0
metadata:
  name: wksp-custom-dwdiu
projects:
  - name: che-theia
    source:
      location: 'https://github.com/eclipse/che-theia.git'
      type: git
      branch: master
components:
  - type: chePlugin
    reference: 'https://raw.githubusercontent.com/eclipse/che-plugin-registry/master/v3/plugins/ms-vscode/vscode-github-pullrequest/0.20.0/meta.yaml'
  - type: cheEditor
    reference: 'https://raw.githubusercontent.com/chepullreq4/pr-check-files/master/che-theia/pr-906/simple/che-theia-editor.yaml'
```
3. Sign-in to the GitHub Pull Requests plugin.
4. Checkout to a new branch and commit some changes.
5. Open a new Pull Request, see https://code.visualstudio.com/docs/editor/github#_creating-pull-requests

The GitHub Pull Requests plugin publishes the branch to origin, so an additional GitHub authentication session is requested.

### PR Checklist

[As the author of this Pull Request I made sure that:](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#pull-request-template-and-its-checklist)


### Reviewers

Reviewers, please comment how you tested the PR when approving it.

### Happy Path Channel

HAPPY_PATH_CHANNEL=stable
